generate_directory_indexes.py

```diff
@@ -115,47 +159,133 @@ def format_directory_listing(directory_listing):
     })
     return out
 
-def walk(path, base_path):
-    contents = os.listdir(path)
-    directory_listing = []
+
+def file_information(full_path, file, last_modified = None, size = None):
```

Comment:
Don't insert spaces between the `=` in argument lists when setting argument defaults.